src/AutoMapper/MappingEngine.cs

```diff
@@ -51,5 +49,72 @@ public object Map(ResolutionContext context)
                 throw new AutoMapperMappingException(context, ex);
             }
         }
+
+        private object Map(object source, ResolutionContext context)

```

Comment:
It certainly could, I think that's the next step towards pre-wiring a mapping plan for a type map. Good call.